llvm/include/llvm/ADT/GenericCycleInfo.h

```diff
@@ -74,12 +74,16 @@ template <typename ContextT> class GenericCycle {
   ///       always have the same depth.
   unsigned Depth = 0;
 
+  /// Cache for the results of GetExitBlocks
+  std::unique_ptr<SmallVector<BlockT *, 4>> ExitBlocksCache;
```

Comment:
Doesn't need to be unique_ptr

Reply:
Same as https://github.com/llvm/llvm-project/pull/112290#discussion_r1802146978